core/src/tasks/profile/destroy.ts

```diff
@@ -0,0 +1,42 @@
+import { CLSTask } from "../../classes/tasks/clsTask";
+import { Export } from "../../models/Export";
+import { Profile } from "../../models/Profile";
+
+export class ProfileDestroy extends CLSTask {
+  constructor() {
+    super();
+    this.name = "profile:destroy";
+    this.description =
+      "Export and destroy profiles that no longer have any directly mapped properties";
+    this.frequency = 0;
+    this.queue = "profiles";
+    this.inputs = {
+      profileId: { required: true },
+    };
+  }
+
+  async runWithinTransaction({ profileId }: { profileId: string }) {
+    const profile = await Profile.findById(profileId);
+    if (!profile) return;
+    if (profile.state !== "ready") return;
+
+    const pendingExports = await Export.count({
+      where: {
+        profileId: profileId,
+        state: ["pending", "processing"],
+      },
+    });
+    if (pendingExports > 0) return;
+
+    const oldGroups = await profile.$get("groups");
+    if (oldGroups.length > 0) {
+      // clear groups and export
+      // when the export is done, this task will be enqueued again to destroy it
+      await Profile.destroyGroupMembers(profile);
+      await profile.export(false, oldGroups, true, false);
```

Comment:
It's good this happens in a transaction - otherwise we could end up in a loop where we keep exporting the profile

Reply:
The last argument to `profile.export()` is kind of interesting here - since this task is only considering one profile at a time, you could do the export `sync=true` in this task maybe?

Reply:
Hmm- what would happen if the export takes longer than the `destroy` task frequency and the `profile:destroy` task for this profile is enqueued to run? Would the transaction make the task wait or would it still bail because there's an active export?

Also, a couple differences come to mind between sync/async exporting:
- Export processors currently aren't handled synchronously (there's a story for that in tracker). Using sync exports in this case would still work, but the export processors would be handled async which is a little odd.
- Sync exports throw errors directly, while async auto retries up to whatever limit is set up. 

I'm not sure what the right thing to do would be if this export fails up to a retry limit. Should we delete the profile from Grouparoo anyway? Should we keep retrying? How can we communicate this? Maybe having the `profile:destroy` task fail and become a resque error is clearer as to what happened, but the recurring `destroy` task might keep trying to delete it so something would probably need to be done there as well.

Reply:
I think it is OK to leave this as-is for now, and we can deal with the edge-cases in another story/PR.  I had forgotten about that we can't do sync async exports.

I believe that as currently implemented, if an export fails, it will end up in resque as an error, and then the Profile would get deleted from Grouparoo anyway (as the blocking export would move to the `failed` state).  At least having a Resque failure lets the user know they should double-check the destination.